third_party/txt/src/txt/paragraph.cc

```diff
@@ -984,12 +984,23 @@ Paragraph::PositionWithAffinity Paragraph::GetGlyphPositionAtCoordinate(
 }
 
 Paragraph::Range<size_t> Paragraph::GetWordBoundary(size_t offset) const {
-  // TODO(garyq): Consider punctuation as separate words.
   if (text_.size() == 0)
     return Range<size_t>(0, 0);
-  return Range<size_t>(
-      minikin::getPrevWordBreakForCache(text_.data(), offset + 1, text_.size()),
-      minikin::getNextWordBreakForCache(text_.data(), offset, text_.size()));
+
+  UErrorCode status = U_ZERO_ERROR;
+  std::unique_ptr<icu::BreakIterator> break_iter(
```

Comment:
Break iterators are expensive. We should find a way to cache these somehow.

Reply:
This API is not called during typical text rendering.  It's used to select the word at the cursor in a text editor.